### PR TITLE
Update the documentation

### PR DESCRIPTION
**Pull request by SpacePossum**
Hi all!

I've made a few minor changes to the docs ~to get those back into shape.
However I could use help finishing the `routing.rst`, this describes the `RoutingServiceProvider`.
The provider is powerful and awesome, but I struggle getting any good copy in there.
So if you have time please help out :)~

### CODE REVIEW

File: doc/providers/form.rst [hunk 1]
```diff
@@ -47,29 +47,30 @@ Registering
...
 
     If you are going to use the validation extension with forms, you must also
-    add a dependency to the ``symfony/config`` and ``symfony/translation``
+    add a dependency to the ``symfony/validator`` and ``symfony/config``
     components:
 
     .. code-block:: bash
 
-        composer require symfony/validator symfony/config symfony/translation
+        composer require symfony/validator symfony/config symfony/config
```
Comment (stof) [inline doc/providers/form.rst]: config is duplicated

File: doc/providers/routing.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+* n/a
+
+Services
+--------
+
+* **route_factory**: Creates and returns instances of set 'route_class'
+* **routes_factory**: Creates and returns instances of RouteCollection
+* **routes**: The 'route_factory'
+* **url_generator**: ``UrlGenerator``
+* **request_matcher**: ``RedirectableUrlMatcher``
```
Comment (stof) [inline doc/providers/routing.rst]: this is not consistent with other docs

File: doc/providers/routing.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+----------
+
+* n/a
+
+Services
+--------
+
+* **route_factory**: Creates and returns instances of set 'route_class'
+* **routes_factory**: Creates and returns instances of RouteCollection
+* **routes**: The 'route_factory'
```
Comment (stof) [inline doc/providers/routing.rst]: this sentence looks incomplete

File: doc/providers/routing.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+.. code-block:: php
+
+    $app->register(new Silex\Provider\RoutingServiceProvider());
+
+Usage
+-----
+
+Set 'route_class' to create routes of that type (default `Silex\\Route`)
+using the 'route_factory'.
```
Comment (stof) [inline doc/providers/routing.rst]: why single quotes here ? this does not make sense

File: doc/providers/routing.rst [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+-----------
+
+.. code-block:: php
+
+    $app->register(new Silex\Provider\RoutingServiceProvider());
+
+Usage
+-----
+
+Set 'route_class' to create routes of that type (default `Silex\\Route`)
```
Comment (stof) [inline doc/providers/routing.rst]: should use double backticks. Single backticks are *not* for code blocks in rST

File: doc/providers/routing.rst (hunk 4 above)
Comment (SpacePossum) [inline doc/providers/routing.rst]: >but I struggle getting any good copy in there.
thats why, its a start, and tagged as WIP, please help me out with good copy :)

File: doc/providers/form.rst [hunk 6]
```diff
@@ -133,10 +138,13 @@ form by adding constraints on the fields::
             'constraints' => new Assert\Email()
         ))
         ->add('billing_plan', ChoiceType::class, array(
-            'choices' => array(1 => 'free', 2 => 'small_business', 3 => 'corporate'),
+            'choices' => array('free' => 1, 'small business' => 2, 'corporate' => 3),
```
Comment (GromNaN) [inline doc/providers/form.rst]: That's correct. http://symfony.com/doc/current/reference/forms/types/choice.html#choices

Comment (fabpot) [inline doc/providers/form.rst]: depends on the version of Symfony

**Comment by fabpot** [comment]
Thank you @SpacePossum.

**Pull request merged**